File: Lab 3/Task2.go
package main

import (
  "fmt"
  "os"
)

/* "One has to ensure that rest of the code does not call os.Exit() anywhere,
 * like flag.ExitOnError, log.Fatalf(), etc."
 * -- https://stackoverflow.com/questions/4278293/how-do-i-return-from-func-main-in-go
 *
 * See "Lab 2"/MainFunction.go
 */
func main() {
  os.Exit(task2_main())
}

func task2_main() int {
  var a, b int

  fmt.Println("Input 2 integers:")
  fmt.Scanf("%d %d", &a, &b)

  if b == 0 {
    fmt.Println("b can't be 0.")
    return 1
  }

  fmt.Printf(
    "%d+%d=%d\n" +
    "%d-%d=%d\n" +
    "%d*%d=%d\n" +
    "%d/%d=%d\n" +
    "%d%%%d=%d\n",
    a, b, a + b,
    a, b, a - b,
    a, b, a * b,
    a, b, a / b,
    a, b, a % b)

  return 0
}

File: Lab 3/Task2.rs
fn main() {
    // let (a, b): (i32, i32);
    let mut buf = String::new();

    println!("Input 2 integers:");
    std::io::stdin().read_line(&mut buf).unwrap();

    // Oh boi, no scanf.
    let mut recv = buf.split_whitespace();
    let a = recv.next().unwrap().parse::<i32>().unwrap();
    let b = recv.next().unwrap().parse::<i32>().unwrap();

    if b == 0 {
        println!("b can't be 0.");
        // See: https://stackoverflow.com/questions/24245276/why-does-rust-not-have-a-return-value-in-the-main-function-and-how-to-return-a
        // See: Lab 2/MainFunction.rs
        std::process::exit(1);
    }

    print!(
        "{}+{}={}\n{}-{}={}\n{}*{}={}\n{}/{}={}\n{}%{}={}\n",
        a, b, a + b,
        a, b, a - b,
        a, b, a * b,
        a, b, a / b,
        a, b, a % b,
    );
}
